lib/between_meals/repo/git.rb

```diff
@@ -116,49 +103,37 @@ def files
         @repo.index.map { |x| { :path => x[:path], :status => :created } }
       end
 
-      def status
-        cmd = Mixlib::ShellOut.new(
-          "#{@bin} status --porcelain 2>&1",
-          :cwd => File.expand_path(@repo_path)
-        )
-        cmd.run_command
-        if cmd.exitstatus != 0
-          @logger.error('Something went wrong with git!')
-          @logger.error(cmd.stdout)
-          fail
+      def upstream?(rev, master = 'remotes/trunk')
+        if @cmd.merge(rev, master).stdout == rev
```

Comment:
merge is a thing - and it's very, very different from merge-base - let's call this `merge_base`